cmd/cli/cmd/migrate.go

```diff
@@ -2,33 +2,90 @@ package cmd
 
 import (
 	"fmt"
+	"strings"
 
+	"github.com/AlecAivazis/survey/v2"
 	"github.com/fatih/color"
 	"github.com/pkg/browser"
 	"github.com/spf13/cobra"
+	"golang.org/x/exp/maps"
+	corev1 "k8s.io/api/core/v1"
 
+	"github.com/kubeshop/botkube/internal/cli"
+	"github.com/kubeshop/botkube/internal/cli/heredoc"
 	"github.com/kubeshop/botkube/internal/cli/migrate"
 	"github.com/kubeshop/botkube/internal/cli/printer"
 )
 
+var (
+	compatibleBotkubeVersions = map[string]bool{
+		"v1.0.0": true,
+		"v1.0.1": true,
+		"v1.1.0": true,
+		"v1.2.0": true,
+	}
```

Comment:
Imagine that we release v1.1.1 (actually we already did it). Should we really display a message that "your version is not supported"? Probably not.

I think we should use `semver` package and just compare major and minor versions.
So that we store
```go
minMajorMinorBotkubeVersion  = "1.0"
maxMajorMinorBotkubeVersion = "1.2"
```